packages/jest-cli/src/Runtime/Runtime.js

```diff
@@ -234,6 +234,7 @@ class Runtime {
     if (
       shouldCollectCoverage &&
       !filename.includes(constants.NODE_MODULES) &&
+      !filename.includes(this._config.mocksPattern) &&
```

Comment:
mocksPattern can actually be a regexp string, so this should be `new RegExp(this._config.mocksPattern)` – can we add this to the constructor:

```js
this._mocksPattern = config.mocksPattern ? new RegExp(config.mocksPattern) : null;
```

and then:
```js
!(mocksPattern && mocksPattern.test(filename))
```

?